pkg/planner/core/planbuilder.go

```diff
@@ -1191,15 +1191,19 @@ func getPossibleAccessPaths(ctx base.PlanContext, tableHints *hint.PlanHints, in
 					continue
 				}
 			}
-			path := &util.AccessPath{Index: index}
 			if index.VectorInfo != nil {
 				// Because the value of `TiFlashReplica.Available` changes as the user modify replica, it is not ideal if the state of index changes accordingly.
 				// So the current way to use the vector indexes is to require the TiFlash Replica to be available.
 				if !tblInfo.TiFlashReplica.Available {
 					continue
 				}
+				path := genTiFlashPath(tblInfo)
```

Comment:
I feel that the original logic is different here set up "tablePath. IsCommonHandlePath = true". What does this setting do?  Why is the original logic not needed？

Reply:
A normal index will create the index's ranges.
But the vector index still uses the table range. So we need to correctly set the pk type. So it can generate the correct table range.
For int pk, it will use `[MinInt64, MaxInt64`] as full range. But for other cases, `[MinNotNull, MaxValue]` will be full range.

Reply:
<img width="620" alt="image" src="https://github.com/user-attachments/assets/a6b1b639-0c8e-4b2b-aa96-8932894397e3">
i think if we don't set it here, we will step into here right? which will derive a int range for common handle tiflash path